Classes/Steno.sc

```diff
@@ -415,21 +417,24 @@ Steno {
 			name = name.asSymbol;
 			if(variables[name].isNil) {
 				"new variable as ".post;
+
 				this.filter(name, { |input, controls|
+					// Bus declaration inside synth func restores busses with
+					// correct channel numbers, e.g. when number of channels changed on the fly
 					var bus = Bus.audio(server, numChannels);
 					var in = XFade2.ar(
 						inA: In.ar(bus, numChannels),
-						inB: InFeedback.ar(bus, numChannels) * \feedback.kr.sign,
-						pan: \feedback.kr.abs.linlin(0, 1, -1, 1)
+						inB: Limiter.ar(InFeedback.ar(bus, numChannels), 8, 0.01) * feedback.sign,
+						// only do InFeedback for first appearance of variable
+						pan: (feedback.abs * (controls.index < 1) * 2 - 1)
 					);
-					variables[name].free; variables[name] = bus;
 
 					// \assignment can be increased for feeding in more than one signal
-					Out.ar(bus, input * (\tokenIndex.kr < \assignment.kr(1)));
+					Out.ar(bus, input * (controls.index < \assignment.kr(1)));
 
 					in * controls[\env] + input
-				})
-
+				});
+				variables[name] = bus;
```

Comment:
You need to free any old one, otherwise you get dangling busses.

Like this:
```
variables[name].free; 
variables[name] = bus;
```

Reply:
since `variables[name]` is `nil` (see conditional statement in line 418), this is not needed. It is not possible to redefine variables (we might need to discuss about this at a later point...)